sci-physics/reduze/files/reduze-2.0.9-gcc6.patch

```diff
@@ -0,0 +1,58 @@
+--- a/reduze/amplitude.cpp
++++ b/reduze/amplitude.cpp
+@@ -782,7 +782,7 @@
+ 
+ 	ex resultLeadingNc = result;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Color");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Color");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 		ex res(1);
+ 		for (size_t i = 0; i < it->op(0).nops(); ++i)
+@@ -823,7 +823,7 @@
+ 	timer.restart();
+ 	exmap m;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Lorentz");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Lorentz");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 
+ 		ex res(1);
+--- a/reduze/files.cpp
++++ b/reduze/files.cpp
+@@ -420,7 +420,7 @@
+ 			get_filename_sectormappings(m.integralfamily()->name());
+ 	ofstream file(file_name.c_str());
+ 	if (!file)
+-		ABORT("Can't write to " << file);
```

Comment:
In C++98, streaming a stream object would have the effect of casting it to a `void *` and streaming its address (or `NULL` address if the failbit was set).  This was omitted in C++11.  IMHO, he best practice for debugging purposes is to stream the name of the file instead ("Can't write to test.txt" instead of "Can't write to 0x000000000000").